src/App.tsx

```diff
@@ -16,80 +17,145 @@ export const goodsFromServer = [
+      ...state,
+      sortType: SortType.ALPHABET,
+    }))
+  );
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+  handleLengthSort = () => (
+    this.setState({
+      sortType: SortType.LENGTH,
+    })
+  );
+
+  handleReverse = () => (
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }))
+  );
+
+  handleReset = () => (
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    })
+  );
+
+  getReorderedGoods = (
+    goods: string[],
+    { sortType, isReversed }: ReorderOptions,
+  ) => {
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case 'ALPHABET':
+          return good1.localeCompare(good2);
+
+        case 'LENGTH':
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    return isReversed ? visibleGoods.reverse() : visibleGoods;
+  };
+
+  render() {
+    const visibleGoods = this.getReorderedGoods(
+      goodsFromServer,
+      this.state,
+    );
+
+    const { sortType, isReversed } = this.state;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-success',
+              { 'is-light': sortType !== 'ALPHABET' },
+            )}
+            onClick={this.handleAlphabetSort}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-success',
+              { 'is-light': sortType !== 'LENGTH' },
+            )}
+            onClick={this.handleLengthSort}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-success',
+              { 'is-light': !isReversed },
+            )}
+            onClick={this.handleReverse}
+          >
+            Reverse
+          </button>
+
+          {
+            (sortType !== 'NONE' || isReversed) && (
```

Comment:
```suggestion
          {(sortType !== 'NONE' || isReversed) && (
```

Create a variable to hold this condition